src/NumberGuessingGame.jsx

```diff
@@ -1,77 +1,56 @@
-import React, { Component } from "react";
-import GuessControl from "./GuessControl";
-import GuessMessage from "./GuessMessage";
-import GameOver from "./GameOver";
+import React, { Component, useState } from 'react'
+import GuessControl from './GuessControl'
+import GuessMessage from './GuessMessage'
+import GameOver from './GameOver'
 
-/**
- *
- * Returns a random integer number from 1-100 inclusive
- */
-function getRandomNumber() {
-  return Math.floor(Math.random() * 100) + 1;
-}
-
-const MAX_ATTEMPTS = 5;
+const MAX_ATTEMPTS = 5
+function NumberGuessingGame() {
+  const [numberToGuess, setNumberToGuess] = useState(
+    Math.floor(Math.random() * 100) + 1
+  )
 
-class NumberGuessingGame extends Component {
-  constructor(props) {
-    super(props);
+  const [numberOfGuesses, setNumberOfGuesses] = useState(0)
 
-    this.state = {
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    };
-
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleGuess = this.handleGuess.bind(this);
-    this.handleReset = this.handleReset.bind(this);
-  }
+  const [latestGuess, setLatestGuess] = useState(null)
 
-  handleGuess(guess) {
-    this.setState({
-      latestGuess: guess,
-      numberOfGuesses: this.state.numberOfGuesses + 1,
-    });
+  const handleGuess = (guess) => {
+    setLatestGuess(Number(guess))
+    setNumberOfGuesses((prevGuesses) => prevGuesses + 1)
   }
 
-  handleReset() {
-    this.setState({
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    });
+  const handleReset = () => {
+    setNumberToGuess(Math.floor(Math.random() * 100) + 1)
+    setNumberOfGuesses(0)
+    setLatestGuess(null)
   }
-
-  render() {
-    const isCorrectGuess = this.state.latestGuess === this.state.numberToGuess;
-
-    const isGameOver =
-      isCorrectGuess || this.state.numberOfGuesses === MAX_ATTEMPTS;
-
+  let isCorrectGuess = latestGuess === numberToGuess
+  let isGameOver = isCorrectGuess || numberOfGuesses === MAX_ATTEMPTS
+
+  if (numberOfGuesses >= MAX_ATTEMPTS + 1) {
```

Comment:
This conditional could also be expressed as `numberOfGuesses > MAX_ATTEMPTS`. What you have isn't wrong and will work the same way, but I thought I'd point this out as an alternative option.